src/main/java/ch/njol/skript/util/slot/EquipmentSlot.java

```diff
@@ -189,6 +190,11 @@ public EquipSlot getEquipSlot() {
 
 	@Override
 	public int getIndex() {
+		if (slot == EquipSlot.TOOL) {
+			Entity holder = e.getHolder();
+			if (holder instanceof Player)
+				return ((Player) holder).getInventory().getHeldItemSlot();
+		}
```

Comment:
Can we do this in the constructor rather than dynamically? I'm concerned about the index changing when calling it after a delay

Reply:
I was wondering about this too. I'll take a look into it.

Reply:
The actual tool returned also has the ability to change, so this should actually be fine.